cpp/src/io/parquet/page_enc.cu

```diff
@@ -1100,15 +1100,21 @@ __global__ void __launch_bounds__(128, 8)
       if (t == 0) { s->cur = dst + total_len; }
       if (is_valid) {
         switch (physical_type) {
-          case INT32:
+          case INT32: [[fallthrough]];
           case FLOAT: {
             int32_t v;
-            if (dtype_len_in == 4)
+            if (dtype_len_in == 8)
+              v = s->col.leaf_column->element<int64_t>(val_idx);
```

Comment:
How do we know this won't overflow? If this is a special case just for millisecond timestamps, should we guard it with a check against that type?

Reply:
Right, there's no guarantee that it won't overflow. I've been thinking only in terms of round-trip of Parquet file so I didn't realize this.
What should be done in case of overflow (based on what the rest of cuDF does)?

I don't think we need another check for the exact type here. This is generic code, only depends on the input size. If we ever supported casting in the Parquet writer, this code would cover int64 to int32 casts (overflow issues and all :)).